src/exo/LoopIR_scheduling.py

```diff
@@ -1074,6 +1074,19 @@ def calc_idx(self, idxs):
 
         return new_idxs
 
+    def calc_dim(self, dim):
+        assert dim < len(
+            [w for w in self.win_stmt.rhs.idx if isinstance(w, LoopIR.Interval)]
+        )
+
+        new_dim = 0
+        for w in self.win_stmt.rhs.idx:
+            if isinstance(w, LoopIR.Interval):
+                dim -= 1
+            if dim == -1:
+                return new_dim
+            new_dim += 1
```

Comment:
This is really quite confusing... it would benefit a lot from a comment describing how it works and a specifically targeted test.